hedera-node/src/main/java/com/hedera/services/txns/schedule/ScheduleReadyForExecution.java

```diff
@@ -0,0 +1,104 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ‍
+ */
+
+import com.hedera.services.ledger.HederaLedger;
+import com.hedera.services.legacy.core.jproto.JKey;
+import com.hedera.services.legacy.exception.InvalidAccountIDException;
+import com.hedera.services.sigs.order.HederaSigningOrder;
+import com.hedera.services.sigs.order.SigStatusOrderResultFactory;
+import com.hedera.services.sigs.verification.InvalidPayerAccountException;
+import com.hedera.services.store.schedule.ScheduleStore;
+import com.hederahashgraph.api.proto.java.ScheduleID;
+import com.hederahashgraph.api.proto.java.TransactionBody;
+
+import java.util.List;
+import java.util.Set;
+
+import static com.hedera.services.legacy.core.jproto.JKey.equalUpToDecodability;
+
+public abstract class ScheduleReadyForExecution {
+    public final static SigStatusOrderResultFactory PRE_HANDLE_SUMMARY_FACTORY =
+            new SigStatusOrderResultFactory(false);
+
+    private final HederaLedger ledger;
+    protected final HederaSigningOrder signingOrder;
+    protected final ScheduleStore store;
+
+    protected ScheduleReadyForExecution(
+            HederaLedger ledger,
+            HederaSigningOrder signingOrder,
+            ScheduleStore store) {
+        this.ledger = ledger;
+        this.signingOrder = signingOrder;
+        this.store = store;
+    }
+
+    protected boolean readyForExecution(ScheduleID scheduleID) throws Exception {
+        var schedule = store.get(scheduleID);
+        var signers = schedule.signers();
+
+        var transaction = TransactionBody.parseFrom(schedule.transactionBody());
+        var payerAccount = schedule.payer().toGrpcAccountId();
+
+        if (!ledger.exists(schedule.payer().toGrpcAccountId())) {
+            throw new InvalidPayerAccountException();
+        }
+
+        var payerPubKey = ledger.get(payerAccount).getKey();
+
+        if (!contains(signers, payerPubKey)) {
+            return false;
+        }
+
+        var result = signingOrder.keysForOtherParties(transaction, PRE_HANDLE_SUMMARY_FACTORY);
+        if (result.hasErrorReport()) {
+            var error = result.getErrorReport();
+            if (error.hasAccountId()) {
+                throw new InvalidAccountIDException(error.getAccountId(), new Throwable());
+            } else {
+                throw new Exception(error.toString());
+            }
+        }
+
+        return otherPartiesKeysAreFound(signers, result.getOrderedKeys());
+    }
+
+    private static boolean otherPartiesKeysAreFound(Set<JKey> set, List<JKey> otherPartiesKeys) {
+        for (JKey otherPartyKey: otherPartiesKeys) {
+            if (!contains(set, otherPartyKey)) {
+                return false;
+            }
+        }
+
+        return true;
+    }
+
+    private static boolean contains(Set<JKey> set, JKey key) {
```

Comment:
`signers` here are simple keys (ECDSA) (the first argument), however, the `payerPubKey` or `otherPartiesKeys` may be a complex threshold key f.e. We need to make sure that hierarchical keys are considered. Maybe something similar to `HederaKeyActivation` -> `otherPartySigsAreActive`